README.md

```diff
@@ -1,306 +1,192 @@
-GitHub Actions.
-
-GitHub Actions allows users to select a specific version of the action to use,
-based on release tags. This script simplifies this process by performing the
-following steps:
-
-1. **Retrieving the latest release tag:** The script starts by fetching the most
-   recent SemVer release tag of the current branch, by looking at the local data
-   available in your repository.
-2. **Prompting for a new release tag:** The user is then prompted to enter a new
-   release tag. To assist with this, the script displays the tag retrieved in
-   the previous step, and validates the format of the inputted tag (vX.X.X). The
-   user is also reminded to update the version field in package.json.
-3. **Tagging the new release:** The script then tags a new release and syncs the
-   separate major tag (e.g. v1, v2) with the new release tag (e.g. v1.0.0,
-   v2.1.2). When the user is creating a new major release, the script
-   auto-detects this and creates a `releases/v#` branch for the previous major
-   version.
-4. **Pushing changes to remote:** Finally, the script pushes the necessary
-   commits, tags and branches to the remote repository. From here, you will need
-   to create a new release in GitHub so users can easily reference the new tags
-   in their workflows.
-
-## Dependency License Management
-
-This template includes a GitHub Actions workflow,
-[`licensed.yaml`](./.github/workflows/licensed.yaml), that uses
-[Licensed](https://github.com/licensee/licensed) to check for dependencies with
-missing or non-compliant licenses. This workflow is initially disabled. To
-enable the workflow, follow the below steps.
-
-1. Open [`licensed.yaml`](./.github/workflows/licensed.yaml)
-2. Uncomment the following lines:
-
-   ```yaml
-   # pull_request:
-   #   branches:
-   #     - main
-   # push:
-   #   branches:
-   #     - main
-   ```
-
-3. Save and commit the changes
-
-Once complete, this workflow will run any time a pull request is created or
-changes pushed directly to `main`. If the workflow detects any dependencies with
-missing or non-compliant licenses, it will fail the workflow and provide details
-on the issue(s) found.
-
-### Updating Licenses
-
-Whenever you install or update dependencies, you can use the Licensed CLI to
-update the licenses database. To install Licensed, see the project's
-[Readme](https://github.com/licensee/licensed?tab=readme-ov-file#installation).
-
-To update the cached licenses, run the following command:
-
-```bash
-licensed cache
+```bnf
+<specifier> ::= <tilde> | <caret> | <partial> | <nightly> | <alias>
+<tilde>     ::= "~" <partial>
+<caret>     ::= "^" <partial>
+<partial>   ::= <n> | <n> "." <n> | <n> "." <n> "." <n>
+<nightly>   ::= <n> "." <n> "-nightly" | "nightly"
+<alias>     ::= "lts" | "min"
+<n>         ::= "0" | <positive> <digits>
+<digits>    ::= <digit> | <digit> <digits>
+<digit>     ::= "0" | <positive>
+<positive>  ::= "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
 ```
```

Comment:
Originally I used a Backus-Naur form grammar similar to what [NPM semver used](https://www.npmjs.com/package/semver#range-grammar). However, that grammar isn't even internally consistent so I instead adopted the [grammar syntax used by semver itself](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions).